web-common/src/features/canvas/components/charts/selector.ts

```diff
@@ -180,3 +185,56 @@ export function getTimeDimensionDefinition(
     };
   });
 }
+
+export function validateChartSchema(
+  ctx: StateManagers,
```

Comment:
Here too, ideally this accepted a `metricsView` rather than be bound to Canvas state managers